pkg/apis/enterprise/v3/common_types.go

```diff
@@ -397,7 +409,7 @@ type PhaseInfo struct {
 	// Status of the phase
 	Status AppPhaseStatusType `json:"status,omitempty"`
 	// RetryCount defines the number of retries completed so far
-	RetryCount int32 `json:"retryCount,omitempty"`
+	TryCount uint32 `json:"tryCount,omitempty"`
```

Comment:
Why are we using `TryCount` now vs `RetryCount`? In general, we use RetryCount if we are actually retrying `n` number of times. And other nit comment is to update the comment to reflect whether we are using retryCount or tryCount.
I think fundamentally, try means we only tried once whereas retry means we tried many times, isn't it?

Reply:
Right now, the Retry semantic is broken. When it fails for the first time, we are incrementing RetryCount to 1, similarly for the 2nd time, ... n times. When somebody looks at the CR status, it doesn't look correct. If we want to track just the retry count alone, then we need the additional logic to differentiate the first time try and then retry logic, and also need to do that consciously, as we are updating the CR on every reconcile. Hence, unifying it with just TryCount, which means how many times we are trying, and it implicitly covers the retry count.


(will address the nit comment in my next commit)

Reply:
Hmm, semantically this is not correct right. Re-try is only a second time and onwards.

Reply:
Yeah, that is what I am saying too :). You tried the very first time and you failed and now you are retrying(for the second time) and the counter reflects how many times you retried. Am I still not making sense? :D

Reply:
My point is, say if we are configuring Max. retry count as 1, and when it fails in a phase, the retry count is marked as 1, same is updated in the CR, which means that when somebody dumps the CR, it kind of sounds like we retried one time and still failed. Actually, we never retried.
In addition to this, existing logic also fails, as the Max. the retry count is reached. 
As i mentioned earlier, for making the try vs retry scenarios clean, we need to differentiate if across the places, and it  increases unnecessary complexity. So, it is simple to go with just the try counter.